src/main/kotlin/com/devexperts/openhack2022/cell_sandbox/game/updater/CellUpdater.kt

```diff
@@ -108,10 +59,110 @@ class CellUpdater: Updater {
         }
     }
 
+    private fun eatFoodByCell(
+        oldCell: CellState,
+        it: FoodState,
+        world: World,
+        delta: Double,
+        cell: CellState
+    ) {
+        if (oldCell.center.distance(it.center) < oldCell.radius + it.radius) {
+            val massToEat = min(it.mass, world.settings.maxFoodAbsorbingSpeed * delta)
+            cell.mass += massToEat
+        }
+    }
+
+    private fun moveCell(
+        oldArea: AreaState,
+        connection: CellConnectionState,
+        cell: CellState,
+        oldCell: CellState,
+        delta: Double
+    ) {
+        val partner = oldArea.cells[connection.partnerId]
+        if (partner != null) {
+            val partnerConnection = partner.connections[cell.id]
+            if (partnerConnection != null) {
+                // TODO why exactly 4 times?
```

Comment:
4 is picked as balance between physical stability and computational load

Reply:
ok got it